Alignment/OfflineValidation/plugins/JetHTAnalyzer.cc

```diff
@@ -0,0 +1,377 @@
+
+   private:
+      virtual void beginJob() override;
+      virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
+      virtual void endJob() override;
+
+      // ----------member data ---------------------------
+  
+      edm::InputTag      pvsTag_                        ;
+      edm::EDGetTokenT<reco::VertexCollection> pvsToken_;
+
+      edm::InputTag      tracksTag_                        ;
+      edm::EDGetTokenT<reco::TrackCollection>  tracksToken_;
+
+      edm::InputTag      triggerTag_;
+      edm::EDGetTokenT<edm::TriggerResults>  triggerToken_;
+
+      int printTriggerTable_;
+      double minVtxNdf_;
+      double minVtxWgt_;
+
+      std::vector<double> profilePtBorders_;
+      std::vector<int> iovList_;
+
+      edm::Service<TFileService> outfile_;
+      TH1F* h_ntrks ;
+      TH1F* h_probePt;
+      TH1F* h_probeEta;
+      TH1F* h_probePhi;
+      TH1F* h_probeDxy;
+      TH1F* h_probeDz;
+      TH1F* h_probeDxyErr;
+      TH1F* h_probeDzErr;   
+
+      SmartSelectionMonitor mon;
+
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+JetHTAnalyzer::JetHTAnalyzer(const edm::ParameterSet& iConfig) :
+  pvsTag_           (iConfig.getParameter<edm::InputTag>("vtxCollection")), 
+  pvsToken_         (consumes<reco::VertexCollection>(pvsTag_)), 
+  tracksTag_        (iConfig.getParameter<edm::InputTag>("trackCollection")), 
+  tracksToken_      (consumes<reco::TrackCollection>(tracksTag_)),
+  triggerTag_       (iConfig.getParameter<edm::InputTag>("triggerResults")), 
+  triggerToken_     (consumes<edm::TriggerResults>(triggerTag_)),
+  printTriggerTable_(iConfig.getUntrackedParameter<int>("printTriggerTable")), 
+  minVtxNdf_        (iConfig.getUntrackedParameter<double>("minVertexNdf")), 
+  minVtxWgt_ (iConfig.getUntrackedParameter<double>("minVertexMeanWeight")),
+  profilePtBorders_ (iConfig.getUntrackedParameter<std::vector<double>>("profilePtBorders")),
+  iovList_ (iConfig.getUntrackedParameter<std::vector<int>>("iovList"))
+{
+   //now do what ever initialization is needed
+}
+
+
+JetHTAnalyzer::~JetHTAnalyzer()
+{
+
+   // do anything here that needs to be done at desctruction time
+   // (e.g. close files, deallocate resources etc.)
+
+}
+
+
+//
+// member functions
+//
+
+// ------------ method called for each event  ------------
+void
+JetHTAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  using namespace edm;
+
+  const double cmToum = 10000;
+
+  edm::ESHandle<TransientTrackBuilder>            theB                ;
+  edm::ESHandle<GlobalTrackingGeometry>           theTrackingGeometry ;
+  iSetup.get<GlobalTrackingGeometryRecord>().get(theTrackingGeometry) ;
+  iSetup.get<TransientTrackRecord>().get("TransientTrackBuilder",theB);
+
+  edm::Handle<reco::VertexCollection> vertices; 
+  iEvent.getByToken(pvsToken_, vertices);
```

Comment:
```suggestion
  const auto& vertices = iEvent.get(pvsToken);
```
as per suggestion at https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideEDMGetDataFromEvent#get